iocage_lib/release.py

```diff
@@ -0,0 +1,56 @@
+import os
+import re
+import requests
+
+from iocage_lib.resource import Resource, ListableResource
+from iocage_lib.ioc_fetch import IOCFetch
+
+
+class Release(Resource):
+
+    @property
+    def path(self):
+        return os.path.join(
+            self.zfs.iocroot_path, 'releases', self.name
+        ) if self.zfs.iocroot_path else None
+
+
+class ListableReleases(ListableResource):
+
+    resource = Release
+    path = 'releases'
+
+    def __init__(self, remote=False, eol_check=True):
+        # We should abstract distribution and have eol checks live there in
+        # the future probably plus release should be able to tell if it's eol
+        # or not. Also perhaps we should think of a filter
+        # interface.
+        super().__init__()
+        self.remote = remote
+        self.eol_check = eol_check
+        self.eol_list = []
+        if eol_check and remote:
+            # TODO: Please let's not use this in the future and look at
+            # comments above
+            self.eol_list = IOCFetch.__fetch_eol_check__()
+
+    def __iter__(self):
+        if self.remote:
+            # TODO: Please abstract this in the future
+            req = requests.get(
+                'https://download.freebsd.org/ftp/'
```

Comment:
Isn’t valid for HBSD we should skip this for that

Reply:
@skarekrow I don't want to overload this with checks or improve things which we are going to remove in the future ( double effort ;) ). Moving on i am going to remove this and abstract host/distribution which would cover this. So this is not needed for now and this api is incompatible with HBSD until we abstract this even more.

Reply:
In other words a WIP ;)